### ISSUE TITLE
Tracking issue for self-hosted support

### ISSUE THREAD

**Issue by cedws** [ankitpokhrel/jira-cli#132]
Ah, I was so excited to be able to escape the horrific Jira UI, but alas I see this project doesn't support self-hosted instances. 😢

Not sure what the API differences are but maybe I can lend a hand.

How about a tracking issue?

**Comment by ankitpokhrel** [comment]
Hi @cedws, thank you for opening the tracking issue. 

First thing we need to do is to figure out which of the current features already works in the local installation. I think some of the features should just work with little to no modifications. Once we have the list of possible issues we need to fix, we can visit them one by one.

To start with this first step, we need to make the `jira init` command work. However, this step can be bypassed by writing the config file manually (`$HOME/.jira/.config.yml` by default, we can provide a custom path with `XDG_CONFIG_HOME` env).

```yaml
board:
  id: <board id, eg: 1>
  name: Board # default board
  type: scrum
epic:
  field: customfield_10011 # custom field name for the epic
issue: # issue types, these info can be fetched from this endpoint https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issues/#api-rest-api-3-issue-createmeta-get
  types:
  - id: "10002"
    name: Task
    subtask: false
  - id: "10003"
    name: Sub-task
    subtask: true
  - id: "10001"
    name: Story
    subtask: false
  - id: "10004"
    name: Bug
    subtask: false
  - id: "10000"
    name: Epic
    subtask: false
login: example@company.com # login email
project: PRJ # project key
server: https://example.company.com # server host
```

It would be great if someone can help with this part.

**Comment by ankitpokhrel** [comment]
I did some tests locally in Jira Software v8.17.1. [This PR](https://github.com/ankitpokhrel/jira-cli/pull/136) will allow us to generate config for both Cloud and Local installations.

I realized that v8.17.1 doesn't support [Jira API v3](https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro/). Features that depend on v1/v2 version of the API seem to be working fine. If the same data is available in v2 and v3 versions, we can adjust the codebase to use v2 instead of v3 to make some commands work. But for features that require mandatory v3, it may not be available fully.
